iphone/Classes/TiUIScrollableViewProxy.m

```diff
@@ -73,9 +73,6 @@ -(NSUInteger)viewCount
 
 -(void)setViews:(id)args
 {
-#if defined(TI_USE_AUTOLAYOUT) || defined(TI_USE_KROLL_THREAD)
```

Comment:
Just tested those issues most of which do relate directly here and none of them have reverted to their broken state. We only need the removeFromSuperview to run on the main thread the rest it doesn't matter if they are running on the kroll or mainThread. The reason this fixes the current issue is because once a js object is out the scope in this case the for loop it will get garbage collected as you know. We create a bridge between the objective C object and the js so it doesn't get retained however when all of this was running on the main-thread while we're running the application on the krill-thread, the kroll thread would have finished executing first. Collecting the JS object before the main-thread had a chance to tell the program to retain it.